x/ecocredit/basket/msg_create.go

```diff
@@ -101,3 +101,20 @@ func validateDateCriteria(d *DateCriteria) error {
 	}
 	return nil
 }
+
+// MsgCreateDenom formats denom and display denom:
+// * denom: eco.<m.Exponent><m.CreditTypeAbbrev>.<m.Name>
+// * display denom: eco.<m.Exponent><m.CreditTypeAbbrev>.<m.Name>
+// Returns error if MsgCrete.Exponent is not supported
+func MsgCreateDenom(m *MsgCreate) (string, string, error) {
```

Comment:
reminder to refrain from adding interface methods to proto types 😉

Reply:
It's not a method - it's a function.

Reply:
ah you're right... read it wrong 👍